lib/spack/spack/test/util/spack_yaml.py

```diff
@@ -0,0 +1,70 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import re
+
+import spack.config
+from spack.main import SpackCommand
+
+config_cmd = SpackCommand('config')
+
+
+def get_config_line(pattern, lines):
+    """Get a configuration line that matches a particular pattern."""
+    line = next((l for l in lines if re.search(pattern, l)), None)
+    assert line is not None, 'no such line!'
+    return line
+
+
+def check_blame(element, scope=None):
+    """Get an element from a config file and check its blame information.
```

Comment:
IMO the phrasing

> a config file

is somewhat confusing since this specifically only tests the contents of the `config` section config file. The explanation itself may need some extra detail (in part because it is confusing for one of the sections to be called `config`, although that's not an issue with this PR specifically):

> Test annotations for the `config` section of the configuration.